ocaml/xapi-idl/lib/debug_info.ml

```diff
@@ -23,9 +23,25 @@ let of_string s =
   let open Tracing in
   match String.split_on_char separator s with
   | [log; traceparent] ->
-      let spancontext = SpanContext.of_traceparent traceparent in
+      let trace_context =
+        try
+          let trace_context = Tracing.TraceContext.of_json_string traceparent in
+          match trace_context with
+          | Ok trace_context ->
+              Some trace_context
+          | Error _ ->
+              None
+        with _ ->
+          Some
+            (TraceContext.empty
+            |> TraceContext.with_traceparent (Some traceparent)
+            )
+      in
+      let spancontext =
+        Option.(join (map Tracing.SpanContext.of_trace_context trace_context))
+      in
       let tracing =
-        Option.map (fun tp -> Tracer.span_of_span_context tp log) spancontext
+        Option.map (Fun.flip Tracer.span_of_span_context log) spancontext
       in
       {log; tracing}
```

Comment:
if possible, tracing here should have type TraceContext.t instead of Span.t. Otherwise, let's keep this in mind for a future PR.